### DISCUSSION TITLE
Pass data to other pages without being seen in url params

### DISCUSSION THREAD

**Discussion by Rakeshtagadghar** [vercel/next.js#12966] [Help] [answered]
Hey guys, I like how nextjs has it page routing system with file names. One thing I am trying to figure out is how to send data to another page without it being seen in url params and not using redux.

**Comment by jensmeindertsma** [comment, marked as answer]
You can create a state in the `_app.js` file that will not be unmounted/remounted when you switch pages. You can pass this state down to a page by passing it to the `Component` component you receive as props ( it resembles the page ). If you want to change the state from the page you can pass the `setState` function down to the `Component` too. I've created a basic example:
```js
// `_app.js`
function App({ Component, pageProps }) {
  const [data, setData] = useState({ name: "Johny" })

  return <Component {...pageProps} data={data} setData={setData} />
}

export default App
```

```js
// `index.js` or any page where you want to use this data
const Home = ({data, setData}) => {
 return (
   <div>
      <h1>Name: {data.name}</h1>
      <button onClick={() => setData({ name: "Rick" })}>Set name to Rick</button>
  </div>
 )
}

export default Home;
```

**Comment by Robokishan** [comment]
This may be old thread. but in production i think its not good idea to do stuff in _app instead you can do is that.
create a function
```
import { NextRouter } from "next/router";

export const push = (
  router: NextRouter,
  pathname: string,
  query: Record<string, string>
) => router.push({ pathname, query }, pathname);
```

and call it like this
```
push(router, "/test", {
              lang: "roman",
            });
```

and this is mostly available in all framework so you  don't have to rewrite logic if you change framework.

**Reply by dcode333** [reply]
Is it proper way to do this??

**Comment by bk973** [comment]
I understand this is an old thread. At the time am posting this answer, Context API from React can be used to solve this issue
